packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -405,6 +427,41 @@ abstract class ProjectBase extends Resource implements IProject {
     return this.cannedMetric(CodeBuildMetrics.failedBuildsSum, props);
   }
 
+  /**
+   * Defines a Codestar notification rule triggered when the build project events occurs by you specified.
+   * It very similar to `onEvent` API
+   * You can also use the methods `notifyOnBuildSucceeded` and `notifyOnBuildFailed` to define rules for these specific event occurs.
+   *
+   * @param id The id of the Codestar notification rule
+   * @param options Customization options for Codestar notification rule
+   * @returns Codestar notification rule associated with this build project.
+   */
+  public notifyOnEvent(id: string, options: notifications.NotifyOnEventOptions = {}): notifications.Rule {
+    const rule = new notifications.Rule(this, id, {
+      ...options, source: this,
+    });
+    rule.addTarget(options.target);
+    return rule;
+  }
+
+  /**
+   * Defines a Codestar notification rule which triggers when a build completes successfully.
+   */
+  public notifyOnBuildSucceeded(id: string, options: notifications.NotifyOnEventOptions = {}): notifications.Rule {
+    const rule = this.notifyOnEvent(id, options);
+    rule.addEvents(['codebuild-project-build-state-succeeded']);
```

Comment:
These should not be just random strings. Let's create a module-private enum in this module to hold these.

Reply:
Done.